back/engines/commercial/idea_assignment/app/models/idea_assignment/extensions/user.rb

```diff
@@ -8,8 +8,33 @@ def self.included(base)
                    class_name: 'Project',
                    foreign_key: :default_assignee_id,
                    dependent: :nullify
+
+          after_update :clean_idea_assignments, if: :saved_change_to_roles?
```

Comment:
We are patching callbacks to the `User` model, instead of patching new side effects to `SideFxUserService`.